src/opcua-iiot-result-filter.ts

```diff
@@ -264,26 +284,37 @@ module.exports = (RED: nodered.NodeAPI) => {
       return result
     }
 
-    const filterByReadType = (payload: Todo) => {
-      return {
-        value: payload.value.filter((item: Todo) => {
-          return item.nodeId.toString().includes(this.nodeId)
-        })
+    const filterByReadType = (payload: ReadPayload) => {
+      if (isArray(payload.value))
+        return {
+          value: payload.value.filter((item: Todo) => {
+            return item.nodeId.toString().includes(this.nodeId)
+          })
+        }
+      else {
+        this.error('wrong payload type');
+        return payload
       }
     }
 
-    // Todo: This feels wrong
-    const filterByWriteType = function (msg: Todo): null {
-      return null // has no value
+    const filterByWriteType = function (payload: Todo) {
+      return {
+        value: payload.nodesToWrite.map((item: Todo) => {
```

Comment:
the `: Todo` after `item` should probably be removed. See [this comment](https://github.com/DATATRONiQ/node-red-contrib-iiot-opcua/pull/10/files#r878138426)

Reply:
This needs to remain Todo because payload is still typed with Todo, and map doesn't like receiving an implicit `any` type